backend/app/Chat/User/UserLock.php

```
@@ -0,0 +1,201 @@
+    /**
+     * Attempts to acquire a lock for a user by UUID.
+     * This prevents race conditions in credit operations.
+     *
+     * @param string $userUuid The user's UUID
+     *
+     * @return bool True if lock was acquired, false otherwise
+     */
+    public static function acquireLock(string $userUuid): bool
+    {
+        try {
+            $pdo = Database::getPdoConnection();
```

PHPMD (2.15.0)

Avoid using static access to class '\MythicalDash\Chat\Database' in method 'acquireLock'. (Clean Code Rules)

(StaticAccess)